src/replicache.test.ts

```diff
@@ -436,35 +425,43 @@ test('subscribe with error', async () => {
   cancel();
 });
 
-test.skip('conflicting commits', async () => {
-  // This test does not use pure functions in the mutations. This is of course
-  // not a good practice but it makes testing easier.
-  const ar = resolver();
-  const br = resolver();
-
-  rep = await replicacheForTesting('conflict');
-  const mutA = rep.register('mutA', async (tx, v) => {
-    await tx.put('k', v);
-    await ar.promise;
-  });
-  const mutB = rep.register('mutB', async (tx, v) => {
-    await tx.put('k', v);
-    await br.promise;
-  });
+test('overlapping writes', async () => {
+  async function dbwait(tx: ReadTransaction, dur: number) {
+    // Try to take setTimeout away from me???
+    const t0 = Date.now();
+    while (Date.now() - t0 > dur) {
+      await tx.get('foo');
+    }
+  }
 
-  // Start A and B at the same commit.
-  const resAFuture = mutA('a');
-  const resBFuture = mutB('b');
+  async function timerwait(dur: number) {
+    await new Promise(res => setTimeout(res, dur));
+  }
 
-  // Finish A.
-  ar.resolve();
-  await resAFuture;
-  expect(await rep.get('k')).to.equal('a');
+  // writes wait on writes
+  rep = await replicacheForTesting('conflict');
+  const mut = rep.register(
+    'wait-then-return',
+    async (tx, {duration, ret}: any) => {
```

Comment:
You can do it!

Reply:
I did it for you

Reply:
Yeah I just can't keep all the ts stuff straight w/o more time in it. Thanks.